src/test/java/org/cactoos/set/SortedTest.java

```diff
@@ -125,4 +125,75 @@ public void mustNotBeEqualToSortedSet() {
                 ))
         ).affirm();
     }
+
+    @Test
+    public void returnsCorrectComparator() {
+        final Comparator<Integer> comparator = Integer::compareTo;
+        new Assertion<>(
+            "Comparator must be the same",
+            new Sorted<>(comparator, 1, 2, 3).comparator(),
+            new IsEqual<>(comparator)
+        ).affirm();
+    }
+
+    @Test
+    public void returnsSubset() {
+        new Assertion<>(
+            "Must return sorted subset",
+            new Sorted<>(Integer::compareTo, 3, 6, 1, 9, 3).subSet(3, 9),
+            new IsIterableContainingInOrder<>(
```

Comment:
@vzurauskas I think you can use directly `IsEqual` with a new `SetOf`.

Reply:
@victornoel It will ignore order, since normal sets are unordered. I need to make sure the order is right here.

Reply:
@vzurauskas ah yes of course!